pkg/storage/storage.go

```diff
@@ -266,28 +262,61 @@ func (s *storageModule) initialize(policy pkg.StoragePolicy) error {
 		}
 	}
 
-	// make sure new pools are mounted
-	log.Info().Msgf("Making sure new volumes are mounted")
-	for idx := range newPools {
-		if _, mounted := newPools[idx].Mounted(); !mounted {
-			log.Debug().Msgf("Mounting volume %s", newPools[idx].Name())
-			if _, err = newPools[idx].Mount(); err != nil {
-				s.brokenPools = append(s.brokenPools, pkg.BrokenPool{Label: newPools[idx].Name(), Err: err})
-				continue
-			}
-			s.volumes = append(s.volumes, newPools[idx])
+	// make sure new pools are added to the list
+	for _, pool := range newPools {
+		_, err := pool.Mount()
+		if err != nil {
+			return err
 		}
+		s.pools = append(s.pools, pool)
 	}
 
 	if err := filesystem.Partprobe(ctx); err != nil {
 		return err
 	}
 
-	return s.ensureCache()
+	if err := s.ensureCache(); err != nil {
+		log.Error().Err(err).Msg("Error ensuring cache")
+		return err
+	}
+
+	if err := s.shutdownUnusedPools(); err != nil {
+		log.Error().Err(err).Msg("Error shutting down unused pools")
+	}
+
+	return nil
+}
+
+func (s *storageModule) shutdownUnusedPools() error {
+	for _, pool := range s.pools {
+		if _, mounted := pool.Mounted(); mounted {
+			volumes, err := pool.Volumes()
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to retrieve subvolumes on pool %s", pool.Name())
+			}
+			log.Debug().Msgf("Pool %s has: %d subvolumes", pool.Name(), len(volumes))
+
+			if len(volumes) > 0 {
+				continue
+			}
+			err = pool.UnMount()
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to unmount volume %s", pool.Name())
+			}
+		}
+		if err := pool.Shutdown(); err != nil {
+			log.Error().Err(err).Msgf("Error shutting down pool %s", pool.Name())
+		}
+	}
+	return nil
 }
 
 func (s *storageModule) Maintenance() error {
-	for _, pool := range s.volumes {
+	for _, pool := range s.pools {
+		// no need to run maintenance on unmounted pools
```

Comment:
Does maintenance is then called whenever we mount the pool again ?

Reply:
No, but maybe we should!